### DISCUSSION TITLE
Could not resolve host: airflow-service-mpack.tar.gz; Name or service not known

### DISCUSSION THREAD

**Discussion by haidernaveed** [apache/airflow#32736] [General]
Hi Team,

I want to install Airflow via Ambari , im using the following guidelines ;
https://github.com/sburn/ambari-airflow

however when I run : ambari-server install-mpack --mpack=airflow-mpack.tar.gz
I get the following error ;Using python  /usr/bin/python
Installing management pack
ERROR: Download airflow-service-mpack.tar.gz with python lib [urllib2] failed with error: (<type 'exceptions.ValueError'>, ValueError('unknown url type: airflow-service-mpack.tar.gz',), <traceback object at 0x7f63d6effb00>)
Trying to download airflow-service-mpack.tar.gz to /var/lib/ambari-server/data/tmp/airflow-service-mpack.tar.gz with [curl] command.
ERROR: Download file airflow-service-mpack.tar.gz with [curl] command failed with error:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: airflow-service-mpack.tar.gz; Name or service not known

ERROR: Unable to download file airflow-service-mpack.tar.gz!
ERROR: unable to donwload file airflow-service-mpack.tar.gz!
ERROR: Management pack could not be downloaded!
ERROR: Exiting with exit code -1.
REASON: Management pack could not be downloaded!

**Comment by Adaverse** [comment]
try below instead - 
```
ambari-server install-mpack --mpack=https://github.com/sburn/ambari-airflow/raw/master/airflow-mpack.tar.gz
```

**Reply by haidernaveed** [reply]
I tried the above command but issue persist , looks like a ssl issue
wget https://github.com/sburn/ambari-airflow/raw/master/airflow-mpack.tar.gz
--2023-07-21 09:59:09--  https://github.com/sburn/ambari-airflow/raw/master/airflow-mpack.tar.gz
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
Unable to establish SSL connection.


How we can resolve this ?

**Reply by Adaverse** [reply]
U can skip certificate check - 
```
wget --no-check-certificate https://github.com/sburn/ambari-airflow/raw/master/airflow-mpack.tar.gz
```

**Reply by haidernaveed** [reply]
ambari-server install-mpack --mpack=https://github.com/sburn/ambari-airflow/raw/master/airflow-mpack.tar.gz --no-check-certificate
Using python  /usr/bin/python
Installing management pack
Usage: ambari-server.py action [options]

ambari-server.py: error: no such option: --no-check-certificate
[root@hdpmn01d-shr-23 (dw) hakram] # wget --no-check-certificate https://github.com/sburn/ambari-airflow/raw/master/airflow-mpack.tar.gz
--2023-07-24 11:21:22--  https://github.com/sburn/ambari-airflow/raw/master/airflow-mpack.tar.gz
Resolving github.com (github.com)... 140.82.114.4
Connecting to github.com (github.com)|140.82.114.4|:443... connected.
Unable to establish SSL connection.

@Adaverse  is ther any other way to integrate Airflow with Ambari ?

**Reply by Adaverse** [reply]
You can try other methods to download the mpack.tar.gz into your machine like curl or lots of other tools.

**Reply by haidernaveed** [reply]
@Adaverse  I have download the package but in order for it to appear in Ambari to Add as a Service which should i place it ?

**Reply by haidernaveed** [reply]
@Adaverse  that was helpful

I can see an option to add service in Ambari now, however while installing im getting this issue :

Traceback (most recent call last):
  File "/var/lib/ambari-agent/cache/stacks/HDP/3.0/services/AIRFLOW/package/scripts/airflow_scheduler_control.py", line 153, in <module>
    AirflowScheduler().execute()
  File "/usr/lib/ambari-agent/lib/resource_management/libraries/script/script.py", line 352, in execute
    method(env)
  File "/var/lib/ambari-agent/cache/stacks/HDP/3.0/services/AIRFLOW/package/scripts/airflow_scheduler_control.py", line 29, in install
    sudo=True)
  File "/usr/lib/ambari-agent/lib/resource_management/core/base.py", line 166, in __init__
    self.env.run()
  File "/usr/lib/ambari-agent/lib/resource_management/core/environment.py", line 160, in run
    self.run_action(resource, action)
  File "/usr/lib/ambari-agent/lib/resource_management/core/environment.py", line 124, in run_action
    provider_action()
  File "/usr/lib/ambari-agent/lib/resource_management/core/providers/system.py", line 263, in action_run
    returns=self.resource.returns)
  File "/usr/lib/ambari-agent/lib/resource_management/core/shell.py", line 72, in inner
    result = function(command, **kwargs)
  File "/usr/lib/ambari-agent/lib/resource_management/core/shell.py", line 102, in checked_call
    tries=tries, try_sleep=try_sleep, timeout_kill_strategy=timeout_kill_strategy, returns=returns)
  File "/usr/lib/ambari-agent/lib/resource_management/core/shell.py", line 150, in _call_wrapper
    result = _call(command, **kwargs_copy)
  File "/usr/lib/ambari-agent/lib/resource_management/core/shell.py", line 314, in _call
    raise ExecutionFailed(err_msg, code, out, err)
resource_management.core.exceptions.ExecutionFailed: Execution of 'python3 -m pip install --upgrade '' pip wheel setuptools' returned 1. ERROR: Invalid requirement: ''
